pkg/engine/variables/vars.go

```diff
@@ -308,82 +308,149 @@ func DefaultVariableResolver(ctx context.EvalInterface, variable string) (interf
 	return ctx.Query(variable)
 }
 
-func substituteVariablesIfAny(log logr.Logger, ctx context.EvalInterface, vr VariableResolver) jsonUtils.Action {
-	isDeleteRequest := isDeleteRequest(ctx)
-	return jsonUtils.OnlyForLeafsAndKeys(func(data *jsonUtils.ActionData) (interface{}, error) {
-		value, ok := data.Element.(string)
-		if !ok {
-			return data.Element, nil
-		}
+func find_next_variable(s string) []int {
```

Comment:
To minimize changes, can we not simply add an `isShallow` check for the shallow identifier (`{{-`) and break out of the loop (i.e. not `continue`) at line 375?